src/VisualStudio/TestUtilities/IntegrationHelper.cs

```diff
@@ -0,0 +1,241 @@
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using EnvDTE;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.Win32;
+using Roslyn.VisualStudio.Test.Utilities.Interop;
+
+using Process = System.Diagnostics.Process;
+
+namespace Roslyn.VisualStudio.Test.Utilities
+{
+    internal static class IntegrationHelper
+    {
+        public static void CreateDirectory(string path, bool deleteExisting = false)
+        {
+            if (deleteExisting)
+            {
+                DeleteDirectoryRecursively(path);
+            }
+
+            Directory.CreateDirectory(path);
+        }
+
+        public static void DeleteDirectoryRecursively(string path)
+        {
+            if (Directory.Exists(path))
+            {
+                DirectoryExtensions.DeleteRecursively(path);
+            }
+        }
+
+        public static async Task DownloadFileAsync(string downloadUrl, string fileName)
+        {
+            using (var webClient = new WebClient())
+            {
+                await webClient.DownloadFileTaskAsync(downloadUrl, fileName).ConfigureAwait(continueOnCapturedContext: false);
+            }
+        }
+
+        public static IntPtr GetModalWindowFromParentWindow(IntPtr parentWindow)
+        {
+            foreach (var topLevelWindow in GetTopLevelWindows())
+            {
+                if ((User32.GetParent(topLevelWindow) == parentWindow) ||
+                    (User32.GetWindow(topLevelWindow, User32.GW_OWNER) == parentWindow) ||
+                    (User32.GetAncestor(topLevelWindow, User32.GA_PARENT) == parentWindow))
+                {
+                    return topLevelWindow;
+                }
+            }
+
+            return IntPtr.Zero;
+        }
+
+        public static string GetTitleForWindow(IntPtr window)
+        {
+            var titleLength = User32.SendMessage(window, User32.WM_GETTEXTLENGTH, IntPtr.Zero, IntPtr.Zero);
+
+            if (titleLength == IntPtr.Zero)
+            {
+                return string.Empty;
+            }
+
+            var title = new StringBuilder(titleLength.ToInt32() + 1);
+
+            User32.SendMessage(window, User32.WM_GETTEXT, (IntPtr)(title.Capacity), title);
+            return title.ToString();
+        }
+
+        public static IEnumerable<IntPtr> GetTopLevelWindows()
+        {
+            var topLevelWindows = new List<IntPtr>();
+
+            var enumFunc = new User32.WNDENUMPROC((hWnd, lParam) => {
+                topLevelWindows.Add(hWnd);
+                return true;
+            });
+            User32.EnumWindows(enumFunc, IntPtr.Zero);
+
+            return topLevelWindows;
+        }
+
+        public static void KillProcess(Process process)
+        {
+            if ((process != null) && (!process.HasExited))
```

Comment:
Unnecessary parenthesis. And it seems odd that we wouldn't throw if you passed null, since that probably means the caller screwed up.